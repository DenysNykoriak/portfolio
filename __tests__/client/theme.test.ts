import { act, cleanup, renderHook } from "@testing-library/react";
import {
  colorTokens,
  themeSettings,
  useThemeInitSettings,
} from "../../client/theme";

describe("Color Tokens", () => {
  test("dark theme snapshot", () => {
    const tokens = colorTokens("dark");

    expect(tokens).toMatchSnapshot("dark-theme-tokens");
  });

  test("light theme snapshot", () => {
    const tokens = colorTokens("light");

    expect(tokens).toMatchSnapshot("light-theme-tokens");
  });
});

describe("Theme Settings", () => {
  test("dark theme snapshot", () => {
    const settings = themeSettings("dark");

    expect(settings).toMatchSnapshot("dark-theme-settings");
  });

  test("light theme snapshot", () => {
    const settings = themeSettings("light");

    expect(settings).toMatchSnapshot("light-theme-settings");
  });
});

describe("useThemeInitSettings Hook", () => {
  afterEach(() => {
    jest.clearAllMocks();
    window.localStorage.clear();
    cleanup();
  });

  test("theme snapshot", () => {
    const { result } = renderHook(useThemeInitSettings);

    expect(result.current.theme).toMatchSnapshot("theme");
  });

  test("test init theme mode (default should be dark) and localstorage key", () => {
    //Dark
    const { result: firstResult } = renderHook(useThemeInitSettings);
    expect(firstResult.current.themeCtxValue.themeMode).toBe("dark");
    cleanup();

    //Light
    window.localStorage.setItem("themeMode", "light");
    const { result: secondResult } = renderHook(useThemeInitSettings);
    expect(secondResult.current.themeCtxValue.themeMode).toBe("light");
  });

  test("should theme toggle", () => {
    const { result } = renderHook(useThemeInitSettings);

    const oldThemeMode = result.current.themeCtxValue.themeMode;

    act(() => {
      result.current.themeCtxValue.toggleThemeMode();
    });

    expect(result.current.themeCtxValue.themeMode).not.toBe(oldThemeMode);
  });

  test("should theme changing", () => {
    const { result } = renderHook(useThemeInitSettings);

    //First Change
    let oldThemeMode = result.current.themeCtxValue.themeMode;
    act(() => {
      result.current.themeCtxValue.setThemeMode(
        oldThemeMode === "dark" ? "light" : "dark"
      );
    });
    expect(result.current.themeCtxValue.themeMode).not.toBe(oldThemeMode);

    //Second Change
    oldThemeMode = result.current.themeCtxValue.themeMode;
    act(() => {
      result.current.themeCtxValue.setThemeMode(
        oldThemeMode === "dark" ? "light" : "dark"
      );
    });
    expect(result.current.themeCtxValue.themeMode).not.toBe(oldThemeMode);
  });
});
