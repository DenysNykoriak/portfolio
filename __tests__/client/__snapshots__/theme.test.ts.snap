// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Tokens dark theme snapshot: dark-theme-tokens 1`] = `
{
  "common": {
    "black": "#17202A",
    "white": "#FBFCFC",
  },
  "darkTheme": {
    "bg": {
      "primary": "#262828",
      "second": "#1e2020",
    },
    "text": "#ECF0F1",
  },
  "lightTheme": {
    "bg": {
      "primary": "#F8F9F9",
      "second": "#e9e9e9",
    },
    "text": "#000000",
  },
  "nowTheme": {
    "bg": {
      "primary": "#262828",
      "second": "#1e2020",
    },
    "text": "#ECF0F1",
  },
  "palette": {
    "gray": {
      "100": "#f2f4f4",
      "200": "#e5e9ea",
      "300": "#d9dfdf",
      "400": "#ccd4d5",
      "500": "#bfc9ca",
      "600": "#99a1a2",
      "700": "#737979",
      "800": "#4c5051",
      "900": "#262828",
    },
    "primary": {
      "100": "#ffe5dc",
      "200": "#ffccb9",
      "300": "#ffb296",
      "400": "#ff9973",
      "500": "#ff7f50",
      "600": "#cc6640",
      "700": "#994c30",
      "800": "#663320",
      "900": "#331910",
    },
    "secondary": {
      "100": "#f8d6e0",
      "200": "#f2adc1",
      "300": "#eb83a1",
      "400": "#e55a82",
      "500": "#de3163",
      "600": "#b2274f",
      "700": "#851d3b",
      "800": "#591428",
      "900": "#2c0a14",
    },
  },
}
`;

exports[`Color Tokens light theme snapshot: light-theme-tokens 1`] = `
{
  "common": {
    "black": "#17202A",
    "white": "#FBFCFC",
  },
  "darkTheme": {
    "bg": {
      "primary": "#262828",
      "second": "#1e2020",
    },
    "text": "#ECF0F1",
  },
  "lightTheme": {
    "bg": {
      "primary": "#F8F9F9",
      "second": "#e9e9e9",
    },
    "text": "#000000",
  },
  "nowTheme": {
    "bg": {
      "primary": "#F8F9F9",
      "second": "#e9e9e9",
    },
    "text": "#000000",
  },
  "palette": {
    "gray": {
      "100": "#262828",
      "200": "#4c5051",
      "300": "#737979",
      "400": "#99a1a2",
      "500": "#bfc9ca",
      "600": "#ccd4d5",
      "700": "#d9dfdf",
      "800": "#e5e9ea",
      "900": "#f2f4f4",
    },
    "primary": {
      "100": "#331910",
      "200": "#663320",
      "300": "#994c30",
      "400": "#ff7f50",
      "500": "#FF8558",
      "600": "#ff9973",
      "700": "#ffb296",
      "800": "#ffccb9",
      "900": "#ffe5dc",
    },
    "secondary": {
      "100": "#2c0a14",
      "200": "#591428",
      "300": "#851d3b",
      "400": "#b2274f",
      "500": "#de3163",
      "600": "#e55a82",
      "700": "#eb83a1",
      "800": "#f2adc1",
      "900": "#f8d6e0",
    },
  },
}
`;

exports[`Theme Settings dark theme snapshot: dark-theme-settings 1`] = `
{
  "palette": {
    "background": {
      "default": "#262828",
      "paper": "#1e2020",
    },
    "mode": "dark",
    "neutral": {
      "dark": "#737979",
      "light": "#e5e9ea",
      "main": "#ccd4d5",
    },
    "primary": {
      "dark": "#994c30",
      "light": "#ffccb9",
      "main": "#ff9973",
    },
    "secondary": {
      "dark": "#851d3b",
      "light": "#f2adc1",
      "main": "#e55a82",
    },
  },
  "typography": {
    "fontFamily": "Libre_Franklin,sans-serif",
    "fontSize": 14,
    "h1": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 40,
    },
    "h2": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 32,
    },
    "h3": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 24,
    },
    "h4": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 20,
    },
    "h5": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 16,
    },
    "h6": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 14,
    },
  },
}
`;

exports[`Theme Settings light theme snapshot: light-theme-settings 1`] = `
{
  "palette": {
    "background": {
      "default": "#F8F9F9",
      "paper": "#e9e9e9",
    },
    "mode": "light",
    "neutral": {
      "dark": "#d9dfdf",
      "light": "#4c5051",
      "main": "#99a1a2",
    },
    "primary": {
      "dark": "#ffb296",
      "light": "#663320",
      "main": "#ff7f50",
    },
    "secondary": {
      "dark": "#eb83a1",
      "light": "#591428",
      "main": "#b2274f",
    },
  },
  "typography": {
    "fontFamily": "Libre_Franklin,sans-serif",
    "fontSize": 14,
    "h1": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 40,
    },
    "h2": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 32,
    },
    "h3": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 24,
    },
    "h4": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 20,
    },
    "h5": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 16,
    },
    "h6": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 14,
    },
  },
}
`;

exports[`useThemeInitSettings Hook theme snapshot: theme 1`] = `
{
  "breakpoints": {
    "between": [Function],
    "down": [Function],
    "keys": [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ],
    "not": [Function],
    "only": [Function],
    "unit": "px",
    "up": [Function],
    "values": {
      "lg": 1200,
      "md": 900,
      "sm": 600,
      "xl": 1536,
      "xs": 0,
    },
  },
  "components": {},
  "direction": "ltr",
  "mixins": {
    "toolbar": {
      "@media (min-width:0px)": {
        "@media (orientation: landscape)": {
          "minHeight": 48,
        },
      },
      "@media (min-width:600px)": {
        "minHeight": 64,
      },
      "minHeight": 56,
    },
  },
  "palette": {
    "action": {
      "activatedOpacity": 0.24,
      "active": "#fff",
      "disabled": "rgba(255, 255, 255, 0.3)",
      "disabledBackground": "rgba(255, 255, 255, 0.12)",
      "disabledOpacity": 0.38,
      "focus": "rgba(255, 255, 255, 0.12)",
      "focusOpacity": 0.12,
      "hover": "rgba(255, 255, 255, 0.08)",
      "hoverOpacity": 0.08,
      "selected": "rgba(255, 255, 255, 0.16)",
      "selectedOpacity": 0.16,
    },
    "augmentColor": [Function],
    "background": {
      "default": "#262828",
      "paper": "#1e2020",
    },
    "common": {
      "black": "#000",
      "white": "#fff",
    },
    "contrastThreshold": 3,
    "divider": "rgba(255, 255, 255, 0.12)",
    "error": {
      "contrastText": "#fff",
      "dark": "#d32f2f",
      "light": "#e57373",
      "main": "#f44336",
    },
    "getContrastText": [Function],
    "grey": {
      "100": "#f5f5f5",
      "200": "#eeeeee",
      "300": "#e0e0e0",
      "400": "#bdbdbd",
      "50": "#fafafa",
      "500": "#9e9e9e",
      "600": "#757575",
      "700": "#616161",
      "800": "#424242",
      "900": "#212121",
      "A100": "#f5f5f5",
      "A200": "#eeeeee",
      "A400": "#bdbdbd",
      "A700": "#616161",
    },
    "info": {
      "contrastText": "rgba(0, 0, 0, 0.87)",
      "dark": "#0288d1",
      "light": "#4fc3f7",
      "main": "#29b6f6",
    },
    "mode": "dark",
    "neutral": {
      "dark": "#737979",
      "light": "#e5e9ea",
      "main": "#ccd4d5",
    },
    "primary": {
      "contrastText": "rgba(0, 0, 0, 0.87)",
      "dark": "#994c30",
      "light": "#ffccb9",
      "main": "#ff9973",
    },
    "secondary": {
      "contrastText": "#fff",
      "dark": "#851d3b",
      "light": "#f2adc1",
      "main": "#e55a82",
    },
    "success": {
      "contrastText": "rgba(0, 0, 0, 0.87)",
      "dark": "#388e3c",
      "light": "#81c784",
      "main": "#66bb6a",
    },
    "text": {
      "disabled": "rgba(255, 255, 255, 0.5)",
      "icon": "rgba(255, 255, 255, 0.5)",
      "primary": "#fff",
      "secondary": "rgba(255, 255, 255, 0.7)",
    },
    "tonalOffset": 0.2,
    "warning": {
      "contrastText": "rgba(0, 0, 0, 0.87)",
      "dark": "#f57c00",
      "light": "#ffb74d",
      "main": "#ffa726",
    },
  },
  "shadows": [
    "none",
    "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
    "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
    "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
    "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
    "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
    "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
    "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
    "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
    "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
    "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
    "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
    "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
    "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
    "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
    "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
    "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
    "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
    "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
    "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
    "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
    "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
    "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
    "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
  ],
  "shape": {
    "borderRadius": 4,
  },
  "spacing": [Function],
  "transitions": {
    "create": [Function],
    "duration": {
      "complex": 375,
      "enteringScreen": 225,
      "leavingScreen": 195,
      "short": 250,
      "shorter": 200,
      "shortest": 150,
      "standard": 300,
    },
    "easing": {
      "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
      "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
      "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
      "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
    },
    "getAutoHeightDuration": [Function],
  },
  "typography": {
    "body1": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": "1rem",
      "fontWeight": 400,
      "lineHeight": 1.5,
    },
    "body2": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": "0.875rem",
      "fontWeight": 400,
      "lineHeight": 1.43,
    },
    "button": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": "0.875rem",
      "fontWeight": 500,
      "lineHeight": 1.75,
      "textTransform": "uppercase",
    },
    "caption": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": "0.75rem",
      "fontWeight": 400,
      "lineHeight": 1.66,
    },
    "fontFamily": "Libre_Franklin,sans-serif",
    "fontSize": 14,
    "fontWeightBold": 700,
    "fontWeightLight": 300,
    "fontWeightMedium": 500,
    "fontWeightRegular": 400,
    "h1": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 40,
      "fontWeight": 300,
      "lineHeight": 1.167,
    },
    "h2": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 32,
      "fontWeight": 300,
      "lineHeight": 1.2,
    },
    "h3": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 24,
      "fontWeight": 400,
      "lineHeight": 1.167,
    },
    "h4": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 20,
      "fontWeight": 400,
      "lineHeight": 1.235,
    },
    "h5": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 16,
      "fontWeight": 400,
      "lineHeight": 1.334,
    },
    "h6": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": 14,
      "fontWeight": 500,
      "lineHeight": 1.6,
    },
    "htmlFontSize": 16,
    "overline": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": "0.75rem",
      "fontWeight": 400,
      "lineHeight": 2.66,
      "textTransform": "uppercase",
    },
    "pxToRem": [Function],
    "subtitle1": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": "1rem",
      "fontWeight": 400,
      "lineHeight": 1.75,
    },
    "subtitle2": {
      "fontFamily": "Libre_Franklin,sans-serif",
      "fontSize": "0.875rem",
      "fontWeight": 500,
      "lineHeight": 1.57,
    },
  },
  "unstable_sx": [Function],
  "unstable_sxConfig": {
    "alignContent": {},
    "alignItems": {},
    "alignSelf": {},
    "backgroundColor": {
      "themeKey": "palette",
      "transform": [Function],
    },
    "bgcolor": {
      "cssProperty": "backgroundColor",
      "themeKey": "palette",
      "transform": [Function],
    },
    "border": {
      "themeKey": "borders",
      "transform": [Function],
    },
    "borderBottom": {
      "themeKey": "borders",
      "transform": [Function],
    },
    "borderBottomColor": {
      "themeKey": "palette",
    },
    "borderColor": {
      "themeKey": "palette",
    },
    "borderLeft": {
      "themeKey": "borders",
      "transform": [Function],
    },
    "borderLeftColor": {
      "themeKey": "palette",
    },
    "borderRadius": {
      "style": [Function],
      "themeKey": "shape.borderRadius",
    },
    "borderRight": {
      "themeKey": "borders",
      "transform": [Function],
    },
    "borderRightColor": {
      "themeKey": "palette",
    },
    "borderTop": {
      "themeKey": "borders",
      "transform": [Function],
    },
    "borderTopColor": {
      "themeKey": "palette",
    },
    "bottom": {},
    "boxShadow": {
      "themeKey": "shadows",
    },
    "boxSizing": {},
    "color": {
      "themeKey": "palette",
      "transform": [Function],
    },
    "columnGap": {
      "style": [Function],
    },
    "display": {},
    "displayPrint": {
      "cssProperty": false,
      "transform": [Function],
    },
    "flex": {},
    "flexBasis": {},
    "flexDirection": {},
    "flexGrow": {},
    "flexShrink": {},
    "flexWrap": {},
    "fontFamily": {
      "themeKey": "typography",
    },
    "fontSize": {
      "themeKey": "typography",
    },
    "fontStyle": {
      "themeKey": "typography",
    },
    "fontWeight": {
      "themeKey": "typography",
    },
    "gap": {
      "style": [Function],
    },
    "gridArea": {},
    "gridAutoColumns": {},
    "gridAutoFlow": {},
    "gridAutoRows": {},
    "gridColumn": {},
    "gridRow": {},
    "gridTemplateAreas": {},
    "gridTemplateColumns": {},
    "gridTemplateRows": {},
    "height": {
      "transform": [Function],
    },
    "justifyContent": {},
    "justifyItems": {},
    "justifySelf": {},
    "left": {},
    "letterSpacing": {},
    "lineHeight": {},
    "m": {
      "style": [Function],
    },
    "margin": {
      "style": [Function],
    },
    "marginBlock": {
      "style": [Function],
    },
    "marginBlockEnd": {
      "style": [Function],
    },
    "marginBlockStart": {
      "style": [Function],
    },
    "marginBottom": {
      "style": [Function],
    },
    "marginInline": {
      "style": [Function],
    },
    "marginInlineEnd": {
      "style": [Function],
    },
    "marginInlineStart": {
      "style": [Function],
    },
    "marginLeft": {
      "style": [Function],
    },
    "marginRight": {
      "style": [Function],
    },
    "marginTop": {
      "style": [Function],
    },
    "marginX": {
      "style": [Function],
    },
    "marginY": {
      "style": [Function],
    },
    "maxHeight": {
      "transform": [Function],
    },
    "maxWidth": {
      "style": [Function],
    },
    "mb": {
      "style": [Function],
    },
    "minHeight": {
      "transform": [Function],
    },
    "minWidth": {
      "transform": [Function],
    },
    "ml": {
      "style": [Function],
    },
    "mr": {
      "style": [Function],
    },
    "mt": {
      "style": [Function],
    },
    "mx": {
      "style": [Function],
    },
    "my": {
      "style": [Function],
    },
    "order": {},
    "overflow": {},
    "p": {
      "style": [Function],
    },
    "padding": {
      "style": [Function],
    },
    "paddingBlock": {
      "style": [Function],
    },
    "paddingBlockEnd": {
      "style": [Function],
    },
    "paddingBlockStart": {
      "style": [Function],
    },
    "paddingBottom": {
      "style": [Function],
    },
    "paddingInline": {
      "style": [Function],
    },
    "paddingInlineEnd": {
      "style": [Function],
    },
    "paddingInlineStart": {
      "style": [Function],
    },
    "paddingLeft": {
      "style": [Function],
    },
    "paddingRight": {
      "style": [Function],
    },
    "paddingTop": {
      "style": [Function],
    },
    "paddingX": {
      "style": [Function],
    },
    "paddingY": {
      "style": [Function],
    },
    "pb": {
      "style": [Function],
    },
    "pl": {
      "style": [Function],
    },
    "position": {},
    "pr": {
      "style": [Function],
    },
    "pt": {
      "style": [Function],
    },
    "px": {
      "style": [Function],
    },
    "py": {
      "style": [Function],
    },
    "right": {},
    "rowGap": {
      "style": [Function],
    },
    "textAlign": {},
    "textOverflow": {},
    "textTransform": {},
    "top": {},
    "typography": {
      "cssProperty": false,
      "themeKey": "typography",
    },
    "visibility": {},
    "whiteSpace": {},
    "width": {
      "transform": [Function],
    },
    "zIndex": {
      "themeKey": "zIndex",
    },
  },
  "zIndex": {
    "appBar": 1100,
    "drawer": 1200,
    "fab": 1050,
    "mobileStepper": 1000,
    "modal": 1300,
    "snackbar": 1400,
    "speedDial": 1050,
    "tooltip": 1500,
  },
}
`;
